doc/kernel/code-relocation.rst

```diff
@@ -149,14 +149,18 @@ XIP area.
 
 This example will place the .text section of the ``xip_external_flash.c`` file
 to the ``EXTFLASH`` memory region where it will be executed from (XIP). The
-.data will be relocated as usual into SRAM.
+.data will be relocated as usual into SRAM, since flash in XIP mode does not
+allow random write access and variables do not have to persist.
 
   .. code-block:: none
 
      zephyr_code_relocate(FILES src/xip_external_flash.c LOCATION EXTFLASH_TEXT NOCOPY)
      zephyr_code_relocate(FILES src/xip_external_flash.c LOCATION SRAM_DATA)
 
-Relocating libraries
+You can additionally specify ``NOCOPY_DATA`` if you do not want .data to be relocated into SRAM.
+This can be useful if it is already loaded into SRAM, e.g. by the Linux kernel
+using the `remoteproc framework <https://docs.kernel.org/staging/remoteproc.html>`_.
+
```

Comment:
I'm sort of surprised this isn't a doxygen error. Good catch above

Reply:
The RST files are not processed by Doxygen as far as I understand, but directly by sphinx+docutils: https://docs.zephyrproject.org/latest/contribute/documentation/generation.html

Reply:
Sorry, shinx error.